tracee-rules/engine/engine.go

```diff
@@ -182,25 +182,18 @@ func (engine *Engine) consumeSources(done <-chan bool) {
 					continue
 				}
 
-				pe, err := ToParsedEvent(traceeEvt)
-				if err != nil {
-					engine.logger.Printf("error converting tracee event to OPA ast.Value: %v", err)
-					engine.signaturesMutex.RUnlock()
-					continue
-				}
-
 				eventOrigin := analyzeEventOrigin(traceeEvt)
 				for _, s := range engine.signaturesIndex[types.SignatureEventSelector{Source: "tracee", Name: traceeEvt.EventName, Origin: eventOrigin}] {
-					engine.dispatchEvent(s, pe)
+					engine.dispatchEvent(s, event.(tracee.Event), parsedEvents)
```

Comment:
We're type casting in four different for loops. Not sure if that's optimal, but maybe we could save cast result in local variable?

Reply:
Roger that - fixed